rm/rm-client/src/test/java/org/ow2/proactive/resourcemanager/common/NodeStateTest.java

```diff
@@ -0,0 +1,31 @@
+package org.ow2.proactive.resourcemanager.common;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import org.junit.Test;
+
+
+/**
+ * @author ActiveEon Team
+ * @since 17/01/17
+ */
+public class NodeStateTest {
+
+    @Test
+    public void testParseValidInput() throws Exception {
+        for (NodeState nodeState : NodeState.values()) {
+            assertThat(NodeState.parse(nodeState.toString())).isEqualTo(nodeState);
+        }
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testParseInvalidInput() throws Exception {
+        NodeState.parse("Unknown Input Value");
+    }
+
+    @Test
+    public void testToString() {
+        assertThat(NodeState.FREE.toString()).isNotEqualTo(NodeState.FREE.name());
```

Comment:
This test is important since `toString` is overridden and not expected to return the same value as the enum name. That's especially why the `NodeState` has a `desc` field.

Regarding the missing tests you mention they are already implemented as system tests: 

https://bitbucket.org/activeeon/scheduling-system-tests/pull-requests/29/update-and-add-system-tests-related-to-the/diff